homeassistant/components/ollama/config_flow.py

```diff
@@ -0,0 +1,148 @@
+"""Config flow for Ollama integration."""
+
+from __future__ import annotations
+
+import asyncio
+import logging
+import sys
+from types import MappingProxyType
+from typing import Any
+
+import httpx
+import ollama
+import voluptuous as vol
+
+from homeassistant.config_entries import (
+    ConfigEntry,
+    ConfigFlow,
+    ConfigFlowResult,
+    OptionsFlow,
+)
+from homeassistant.const import CONF_URL
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.selector import (
+    NumberSelector,
+    NumberSelectorConfig,
+    NumberSelectorMode,
+    TemplateSelector,
+    TextSelector,
+    TextSelectorConfig,
+    TextSelectorType,
+)
+
+from .const import (
+    CONF_MAX_HISTORY,
+    CONF_MODEL,
+    CONF_PROMPT,
+    DEFAULT_PROMPT,
+    DEFAULT_TIMEOUT,
+    DOMAIN,
+    MAX_HISTORY_NO_LIMIT,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_URL): TextSelector(
+            TextSelectorConfig(type=TextSelectorType.URL)
+        ),
+        vol.Required(CONF_MODEL): cv.string,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> None:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    client = ollama.AsyncClient(host=data[CONF_URL])
+    async with asyncio.timeout(DEFAULT_TIMEOUT):
+        await client.list()
+
+
+class OllamaConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Ollama."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        errors = {}
+
+        try:
+            await validate_input(self.hass, user_input)
+        except (TimeoutError, httpx.ConnectError):
+            errors["base"] = "cannot_connect"
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            errors["base"] = "unknown"
+        else:
+            return self.async_create_entry(title="Ollama", data=user_input)
```

Comment:
Idea: if a user configures a second Ollama instance, maybe the title can be set to the URL ?

Reply:
I set the title using the model, which is now configured once during the initial set up (and not in the options flow)